src/OnboardingSPA/pages/Steps/DesignHeaderMenu/index.js

```diff
@@ -1,24 +1,90 @@
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { useViewportMatch } from '@wordpress/compose';
+
+import { LivePreview } from '../../../components/LivePreview';
 import CommonLayout from '../../../components/Layouts/Common';
-import StepOverview from '../../../components/StepOverview';
-import { VIEW_DESIGN_HEADER_MENU } from '../../../../constants';
+import { GlobalStylesProvider } from '../../../components/LivePreview';
+import { DesignStateHandler } from '../../../components/StateHandlers';
+
+
+import {
+	VIEW_DESIGN_HEADER_MENU,
+	THEME_STATUS_ACTIVE,
+} from '../../../../constants';
 import { store as nfdOnboardingStore } from '../../../store';
-import { useDispatch } from '@wordpress/data';
-import { useEffect } from '@wordpress/element';
 
 const StepDesignHeaderMenu = () => {
-	const { setDrawerActiveView, setIsDrawerOpened, setIsSidebarOpened } =
-		useDispatch( nfdOnboardingStore );
+	const { 
+		themeStatus,
+		headerMenu,
+	 } = useSelect(
+		( select ) => {
+			return {
+				themeStatus: select( nfdOnboardingStore ).getThemeStatus(),
+				headerMenu: select( nfdOnboardingStore ).getHeaderMenuData(),
+			};
+		},
+		[]
+	);
+
+	const [ isLoaded, setIsLoaded ] = useState( false );
+	const [ pattern, setPattern ] = useState();
+	const isLargeViewport = useViewportMatch( 'medium' );
+
+	const { 
+		setDrawerActiveView, 
+		setIsDrawerOpened, 
+		setIsDrawerSuppressed,
+		setIsSidebarOpened,
+	} = useDispatch( nfdOnboardingStore );
+
+	const getCurrentPattern = async () => {
```

Comment:
Do we need this as this is basically empty? even the isLoaded state and it's useEffect

Reply:
yup ... not needed .. removed the code